docs/adr/0010-migrate-to-v3-cluster-resource.md

```diff
@@ -0,0 +1,66 @@
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+
+- [10. Migrate to v3 cluster resource](#10-migrate-to-v3-cluster-resource)
+  - [Context](#context)
+  - [Decision](#decision)
+  - [Consequences](#consequences)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+# 10. Migrate to v3 cluster resource
+
+- Status: proposed
+- Date: 2024-06-12
+- Authors: @alexander-demicev @salasberryfin
+- Deciders: @Danil-Grigorev @furkatgofurov7 @yiannistri @mjura
+
+## Context
+
+The current implementation of the import process via Turtles is based on creating a `clusters.provisioning.cattle.io` (also referred to as v1 clusters) resource that represents the Rancher cluster. The controller will watch CAPI cluster resources and trigger the creation of a `clusters.provisioning.cattle.io` when a cluster or the cluster's namespace is labeled for import.
+A new controller was introduced recenty that would follow an equivalent procedure for import but would use `clusters.management.cattle.io` (also referred to as v3 clusters) resources to represent Rancher clusters. For now, Turtles continues to use the original controller and the new implementation is sitting behind a disabled feature gate.
```

Comment:
Can this reference previous ADR introducing the controller and the differences this ADR brings? Also the purpose of this ADR to set the FG to true by default. No need to put “For now … disabled feature gate”